src/Workspaces/MSBuild/Core/MSBuild/SolutionFileReader.SolutionFilterReader.cs

```diff
@@ -11,7 +11,7 @@
 
 namespace Microsoft.CodeAnalysis.MSBuild;
 
-public partial class MSBuildProjectLoader
+internal partial class SolutionFileReader
```

Comment:
nit: line 40 mentions that TryGetAbsoluteSolutionPath "should throw", but I don't see why it would ever do that.

Reply:
edit: I see, it's all in the DiagnosticReportingMode.Throw. It feels like when we call the method in this way, it would be reasonable to ignore the result or assert on it.